lib/smart_proxy_openscap/openscap_report_parser.rb

```diff
@@ -1,111 +1,42 @@
 # encoding=utf-8
-require 'openscap'
-require 'openscap/ds/arf'
-require 'openscap/xccdf/testresult'
-require 'openscap/xccdf/ruleresult'
-require 'openscap/xccdf/rule'
-require 'openscap/xccdf/fix'
-require 'openscap/xccdf/benchmark'
-require 'json'
-require 'iconv' if RUBY_VERSION.start_with? '1.8'
+require 'tempfile'
 
 module Proxy::OpenSCAP
   class Parse
-    def initialize(arf_data)
-      OpenSCAP.oscap_init
-      size         = arf_data.size
-      @arf_digest  = Digest::SHA256.hexdigest(arf_data)
-      @arf         = OpenSCAP::DS::Arf.new(:content => arf_data, :path => 'arf.xml.bz2', :length => size)
-      @test_result  = @arf.test_result
-
-      @results      = @test_result.rr
-      @sds          = @arf.report_request
-      bench_source = @sds.select_checklist!
-      @benchmark    = OpenSCAP::Xccdf::Benchmark.new(bench_source)
-      @items        = @benchmark.items
-    end
-
-    def cleanup
-      @test_result.destroy if @test_result
-      @benchmark.destroy if @benchmark
-      @sds.destroy if @sds
-      @arf.destroy if @arf
-      OpenSCAP.oscap_cleanup
-    end
-
-    def as_json
-      parse_report.to_json
-    end
-
-    private
-
-    def parse_report
-      report        = {}
-      report[:logs] = []
-      passed        = 0
-      failed        = 0
-      othered         = 0
-      @results.each do |rr_id, result|
-        next if result.result == 'notapplicable' || result.result == 'notselected'
-        # get rules and their results
-        rule_data = @items[rr_id]
-        report[:logs] << populate_result_data(rr_id, result.result, rule_data)
-        # create metrics for the results
-        case result.result
-          when 'pass', 'fixed'
-            passed += 1
-          when 'fail'
-            failed += 1
-          else
-            othered += 1
-        end
+    include ::Proxy::Log
+    include ::Proxy::Util
+
+    def initialize(cname, policy_id, date)
+      @cname = cname
+      @policy_id = policy_id
+      @date = date
+    end
+
+    def as_json(arf_data)
+      in_file = nil
+      json_file = nil
+      begin
+        in_file = Tempfile.new(filename, "/var/tmp")
+        in_file.write arf_data
+        json_file = Tempfile.new("#{filename}-json-", "/var/tmp")
+
+        shell_command "#{script_location} #{in_file.path} #{json_file.path}"
+        json = File.read(json_file)
+      ensure
```

Comment:
I explicitly call delete because [docs](https://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Explicit+close) claim it to be a good practice.